src/domain/chainContext.ts

```diff
@@ -294,6 +323,26 @@ export class ChainContext {
       }
     }
   }
+
+  get status(): ChainStatus {
+    const { sync, chainId } = this;
+    return {
+      sync,
+      chainId,
+      lastProcessedBlock: this.registry.lastProcessedBlock ?? 0,
+    };
+  }
+
+  get health(): ChainHealth {
+    return {
+      ...this.status,
+      isHealthy: this.isHealthy(),
+    };
+  }
+
+  private isHealthy(): boolean {
+    return this.sync === ChainSync.SYNCED;
```

Comment:
I think this is for the readyness probe, and not for the liveness probe, right?

Reply:
> The kubelet uses readiness probes to know when a container is ready to start accepting traffic. A Pod is considered ready when all of its containers are ready. One use of this signal is to control which Pods are used as backends for Services. When a Pod is not ready, it is removed from Service load balancers.

On this basis, I think we can actually use the same API endpoint (`/health`) to do both the startup probe and the liveness probe. As readiness probes are related to use of the pod within a service definition (to be load balanced again), this isn't a case that we're using, so I don't think we would implement a readiness probe?